xds/internal/balancer/edsbalancer/configbuilder_test.go

```diff
@@ -0,0 +1,164 @@
+		testLocalityIDs = append(testLocalityIDs, internal.LocalityID{Zone: fmt.Sprintf("test-zone-%d", i)})
+		var (
+			addrs []string
+			ends  []xdsclient.Endpoint
+		)
+		for j := 0; j < addressPerLocality; j++ {
+			addr := fmt.Sprintf("addr-%d-%d", i, j)
+			addrs = append(addrs, addr)
+			ends = append(ends, xdsclient.Endpoint{
+				Address:      addr,
+				HealthStatus: xdsclient.EndpointHealthStatusHealthy,
+			})
+		}
+		testAddressStrs = append(testAddressStrs, addrs)
+		testEndpoints = append(testEndpoints, ends)
+	}
+
+	testLocalitiesP0 = []xdsclient.Locality{
+		{
+			Endpoints: testEndpoints[0],
+			ID:        testLocalityIDs[0],
+			Weight:    20,
+			Priority:  0,
+		},
+		{
+			Endpoints: testEndpoints[1],
+			ID:        testLocalityIDs[1],
+			Weight:    80,
+			Priority:  0,
+		},
+	}
+	testLocalitiesP1 = []xdsclient.Locality{
+		{
+			Endpoints: testEndpoints[2],
+			ID:        testLocalityIDs[2],
+			Weight:    20,
+			Priority:  1,
+		},
+		{
+			Endpoints: testEndpoints[3],
+			ID:        testLocalityIDs[3],
+			Weight:    80,
+			Priority:  1,
+		},
+	}
+}
+
+// TestBuildPriorityConfigMarshalled is a sanity check that the generated config
+// bytes are valid (can be parsed back to a config struct).
+//
+// The correctness is covered by the unmarshalled version
+// TestBuildPriorityConfig.
+func TestBuildPriorityConfigMarshalled(t *testing.T) {
+	const (
+		testClusterName     = "cluster-name-for-watch"
+		testEDSServiceName  = "service-name-from-parent"
+		testLRSServer       = "lrs-addr-from-config"
+		testMaxReq          = 314
+		testDropCategory    = "test-drops"
+		testDropOverMillion = 1
+	)
+	for _, lrsServer := range []*string{newString(testLRSServer), newString(""), nil} {
+		got, _, err := buildPriorityConfigJSON(xdsclient.EndpointsUpdate{
+			Drops: []xdsclient.OverloadDropConfig{{
+				Category:    testDropCategory,
+				Numerator:   testDropOverMillion,
+				Denominator: million,
+			}},
+			Localities: []xdsclient.Locality{{
+				Endpoints: testEndpoints[3],
+				ID:        testLocalityIDs[3],
+				Weight:    80,
+				Priority:  1,
+			}, {
+				Endpoints: testEndpoints[1],
+				ID:        testLocalityIDs[1],
+				Weight:    80,
+				Priority:  0,
+			}, {
+				Endpoints: testEndpoints[2],
+				ID:        testLocalityIDs[2],
+				Weight:    20,
+				Priority:  1,
+			}, {
+				Endpoints: testEndpoints[0],
+				ID:        testLocalityIDs[0],
+				Weight:    20,
+				Priority:  0,
+			}}},
+			&EDSConfig{
+				ChildPolicy:                &loadBalancingConfig{Name: roundrobin.Name},
+				ClusterName:                testClusterName,
+				EDSServiceName:             testEDSServiceName,
+				MaxConcurrentRequests:      newUint32(testMaxReq),
+				LrsLoadReportingServerName: lrsServer,
+			},
+		)
+		if err != nil {
+			t.Fatalf("unexpected error when building priority config: %v", err)
+		}
+		priorityB := balancer.Get(priority.Name)
+		_, err = priorityB.(balancer.ConfigParser).ParseConfig(got)
+		if err != nil {
```

Comment:
Combine the assignment and the conditional.

Reply:
Done